### PR TITLE
Add logging for various clan events

### LINKED ISSUE #684: Change UUIDItem log retrieval from last x to pages
Currently, logs are retrieved by the last x logs (default 5). Retrieving large numbers is difficult. Change this to a page based system, like /c list or /c help.

### PR DESCRIPTION
**Pull request by StudentAlleg**
Adds more logging for clan actions. Add infrastructure to store and retrieve important relevant data to show to users. Re-work UUIDItem logger to use a different table for formatted logs

Fixes #684 

- [x] Rework database to not use logs and not use formatted strings
- [x] Add views to represent specific data in retrievable logs
- [x] Change infrastructure to create log meta data cleanly
- [x] Change infrastructure to receive different types of logs and build them
- [x] Add ability to get all logs of a Clan
- [x] Add ability to get the Clan of the player at Time
- [x] Add ability to get all kill/death logs related with the clan
- [x] Add ability to get a list of all players that were in a clan
- [x] Add ability to get a list of all Clans a player has been in
- [x] Add capability to reference disbanded clans in logs
- [x] Add infrastructure to reference disbanded clans
- [x] Change UUIDItems related things to new system
- [x] Implement logs for Clan Events
- [x] Remove Logger
- [x] Change UUIDLogger to be non static and apart of UUIDRepository
- [x] Fix SonarCloud errors
- [x] Add nice formats for clan log types
- [x] Add nice formats for UUID Item log types

- [x] Change from offlinePlayer to retrieving name in database

Future Considerations
- Allow filtering by log type

## Checklist before requesting a review
- [x] I have performed a self-review of my code
- [ ] I have tested my changes.

### CODE REVIEW

**Comment by StudentAlleg** [comment]
Updated scope, code is in a rough state of the idea I want to go in for clanlogmeta and its usages

**Comment by StudentAlleg** [comment]
correctly builds, when removing not yet implemented code (that is causing the build project to fail)
TODO: find all the errors I made

File: clans/src/main/java/me/mykindos/betterpvp/clans/clans/repository/ClanRepository.java [hunk 1]
```diff
@@ -54,6 +54,7 @@ public class ClanRepository implements IRepository<Clan> {
     private final Database database;
     private final PropertyMapper propertyMapper;
 
+
```
Comment (StudentAlleg) [inline clans/src/main/java/me/mykindos/betterpvp/clans/clans/repository/ClanRepository.java:57]:
```suggestion
```

File: clans/src/main/java/me/mykindos/betterpvp/clans/logging/ClanLogger.java [hunk 2]
```diff
@@ -0,0 +1,311 @@
...
+            Clan killerClan = clanManager.getClanByPlayer(killer).orElse(null);
+            Clan victimClan = clanManager.getClanByPlayer(victim).orElse(null);
+            double dominance = 0;
+            if (clanManager.getRelation(killerClan, victimClan).equals(ClanRelation.ENEMY)) {
+                assert killerClan != null;
+                assert victimClan != null;
+                if (!killerClan.isNoDominanceCooldownActive() && !victimClan.isNoDominanceCooldownActive()) {
+                    dominance = clanManager.getDominanceForKill(killerClan.getSquadCount(), victimClan.getSquadCount());
+                }
+            }
```
Comment (StudentAlleg) [inline clans/src/main/java/me/mykindos/betterpvp/clans/logging/ClanLogger.java:89]:
```suggestion
            }
            
```

File: clans/src/main/java/me/mykindos/betterpvp/clans/clans/commands/subcommands/GetClansOfPlayerSubCommand.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            UtilMessage.message(player, "Clan", getUsage());
+            return;
+        }
+
+        clientManager.search().offline(args[0], clientOptional -> {
+            if (clientOptional.isEmpty()) {
+                UtilMessage.message(player, "Clan", "<green>%s</green> is not a valid player", args[0]);
+                return;
+            }
+            Client client1 = clientOptional.get();
```
[the remark this answers is not in the document]
Comment (StudentAlleg) [reply clans/src/main/java/me/mykindos/betterpvp/clans/clans/commands/subcommands/GetClansOfPlayerSubCommand.java]: changed

File: clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClanEventListener.java [hunk 4]
```diff
@@ -308,7 +374,15 @@ public void onMemberKicked(ClanKickMemberEvent event) {
                 targetPlayer.closeInventory();
             }
         }
-
+        UUID id = log.info(">{} ({}) was kicked by {} ({}) from {} ({})", player.getName(), player.getUniqueId(),
```
[the remark this answers is not in the document]
Comment (StudentAlleg) [reply clans/src/main/java/me/mykindos/betterpvp/clans/clans/listeners/ClanEventListener.java]: removed

File: clans/src/main/java/me/mykindos/betterpvp/clans/clans/commands/subcommands/PlayerTimeSubCommand.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+        long time = 0;
+        if (args.length >= 2) {
+            try {
+                time = Long.parseLong(args[1]);
+            } catch (NumberFormatException ignored) {
+                //pass
+            }
+        }
+        long finalTime = time / (60L * 60L * 1000L);
+        clientManager.search().offline(args[0], client1 -> {
```
[the remark this answers is not in the document]
Comment (StudentAlleg) [reply clans/src/main/java/me/mykindos/betterpvp/clans/clans/commands/subcommands/PlayerTimeSubCommand.java]: changed

**Review by Mykindos — APPROVED** [review]

**Pull request merged**